aioinject/providers.py

```diff
@@ -159,34 +166,59 @@ class DependencyLifetime(enum.Enum):
 
 @runtime_checkable
 class Provider(Protocol[_T]):
-    type_: type[_T]
     impl: Any
     lifetime: DependencyLifetime
+    _cached_dependencies: tuple[Dependency[object], ...]
+    _cached_type: type[_T]  # TODO: I think it is redundant.
```

Comment:
What do you think? I think this can be removed. I can find a valid usecase for this

Reply:
You should provide the type in the provider I think no?

Reply:
> You should provide the type in the provider I think no?

Not necessarily, in most cases type is derived from the object you pass into provider, e.g. `Scoped(Test)` would have type `Test`